Add ESLint

It looks like when you run `npm lint` there are some failures, so maybe we should only consider this done when we've fixed those?

```
➜  inactive-contributor-bot git:(main) npm run lint

> action-retire-inactive-contributor@1.0.0 lint
> eslint . --ext .ts


/Users/matt/git/github.com/cucumber/inactive-contributor-bot/features/step_definitions/steps.ts
    7:8   error  Replace `"@cucumber/cucumber"` with `'@cucumber/cucumber'`                                                                                                        prettier/prettier
    8:20  error  Replace `"assert"` with `'assert'`                                                                                                                                prettier/prettier
    9:32  error  Replace `"../../src/retireInactiveContributors"` with `'../../src/retireInactiveContributors'`                                                                    prettier/prettier
   10:44  error  Replace `"../../src/retireInactiveContributors"` with `'../../src/retireInactiveContributors'`                                                                    prettier/prettier
   58:9   error  Replace `"team"` with `'team'`                                                                                                                                    prettier/prettier
   64:9   error  Replace `"user"` with `'user'`                                                                                                                                    prettier/prettier
   68:3   error  Replace `"a·user·{user}·has·write·access·to·the·cucumber-js·repo"` with `'a·user·{user}·has·write·access·to·the·cucumber-js·repo'`                                prettier/prettier
   72:12  error  Replace `"pending"` with `'pending'`                                                                                                                              prettier/prettier
   77:3   error  Replace `"a·user·{user}·is·part/member·of·{team}"` with `'a·user·{user}·is·part/member·of·{team}'`                                                                prettier/prettier
   91:3   error  Replace `"a·user·{user}·is·a·member·of·{team}"` with `'a·user·{user}·is·a·member·of·{team}'`                                                                      prettier/prettier
   96:12  error  Replace `"pending"` with `'pending'`                                                                                                                              prettier/prettier
  100:6   error  Replace `"the·action·runs"` with `'the·action·runs'`                                                                                                              prettier/prettier
  106:3   error  Replace `"{user}·should·be·in·{team}"` with `'{user}·should·be·in·{team}'`                                                                                        prettier/prettier
  117:3   error  Replace `"{user}·should·not·have·any·custom·permissions·on·the·cucumber-js·repo"` with `'{user}·should·not·have·any·custom·permissions·on·the·cucumber-js·repo'`  prettier/prettier
  121:12  error  Replace `"pending"` with `'pending'`                                                                                                                              prettier/prettier
  126:3   error  Replace `"{user}·should·not·be·in·{team}(·anymore)"` with `'{user}·should·not·be·in·{team}(·anymore)'`                                                            prettier/prettier

/Users/matt/git/github.com/cucumber/inactive-contributor-bot/src/OctokitGithub.ts
   5:19  warning  'user' is defined but never used            @typescript-eslint/no-unused-vars
   9:17  warning  'user' is defined but never used            @typescript-eslint/no-unused-vars
   9:31  warning  'alumniTeam' is defined but never used      @typescript-eslint/no-unused-vars
  13:22  warning  'user' is defined but never used            @typescript-eslint/no-unused-vars
  13:36  warning  'committersTeam' is defined but never used  @typescript-eslint/no-unused-vars
  17:16  warning  'team' is defined but never used            @typescript-eslint/no-unused-vars

✖ 22 problems (16 errors, 6 warnings)
  16 errors and 0 warnings potentially fixable with the `--fix` option.
```

--- a/package.json
+++ b/package.json
@@ -11,7 +11,7 @@
   ],
   "scripts": {
     "build": "tsc",
-    "test": "npm run test:unit && npm run test:acceptance",
+    "test": "npm run lint && npm run test:unit && npm run test:acceptance",
     "test:acceptance": "cucumber-js --tags 'not @todo and not @wip'",
     "test:acceptance:wip": "cucumber-js --tags @wip",
     "test:unit": "mocha src/**/*.test.ts",

